ModelScript methods

JaXt0r opened the pull request:
I implemented the full data export for ModelScript.

(Tedious work. 😆 )

Some hints:

* I tried to combine elements wherever a full property set could be exported within one method. e.g.
```c++
PXC_API void pxMdsGetAnimation_EventPfxStop(PxModelScript* mds, uint32_t animIndex, uint32_t pfxStopIndex,
	int32_t* frame,
	int32_t* index);
```

* As there were vectors within vectors, (e.g. ```model_script.animations[].events[]_```) I added an underscore to highlight that it's a sublist of a list. e.g.
```c++
int32_t pxMdsGetAnimation_EventTagFrame(PxModelScript* mds, uint32_t animIndex, uint32_t eventTagIndex);
```

* C# endpoint is developed and works already.


Feel free to suggest improvements

JaXt0r committed:
Added .editorconfig to comply with clang-lint in the future. ;-)

JaXt0r committed:
Initial commit of MeshScript functions.

JaXt0r committed:
Fixed AnimationCombination.

JaXt0r committed:
ModelScript: Changed implementation of getter to separate methods per property (required for Managed endpoints to work properly).

JaXt0r committed:
MeshScript: Skeleton added.

JaXt0r committed:
Full implementation of ModelScripts done.

lmichaelis commented on .editorconfig:
If you feel like an EditorConfig is necessary, please add it in a separate PR.

lmichaelis commented on include/phoenix/cffi/ModelScript.h:
```diff
@@ -14,7 +14,168 @@ typedef phoenix::model_script PxModelScript;
+	forward = 0,  ///< The animation samples are played from first to last.
+	backward = 1, ///< The animation samples are played from last to first.
+} PxAnimationDirection;
+
+typedef enum {
+	unknown,
+	create_item,
+	insert_item,
+	remove_item,
+	destroy_item,
+	place_item,
+	exchange_item,
+	fight_mode,
+	place_munition,
+	remove_munition,
+	draw_sound,
+	undraw_sound,
+	swap_mesh,
+	draw_torch,
+	inventory_torch,
+	drop_torch,
+	hit_limb,
+	hit_direction,
+	dam_multiply,
+	par_frame,
+	opt_frame,
+	hit_end,
+	window,
+} PxEventTagType;
+
+typedef enum {
+	fist,       ///< The player fights with his fists.
+	one_handed, ///< The player wields a one-handed weapon.
+	two_handed, ///< The player wields a two-handed weapon.
+	bow,        ///< The player wields a bow.
+	crossbow,   ///< The player wields a crossbow.
+	magic,      ///< The player casts a magic spell.
+	none,       ///< The player is not in a fighting stance.
+	invalid,    ///< A fight mode which acts as an `unset` marker. Added for OpenGothic compatibility.
+} PxEventFightMode;
```
Enum members must be prefixed with `Px` too. Ideally the naming convention should be for example:

```c
typedef enum {
   PxAnimationDirectionForward = 0,
   PxAnimationDirectionBackward = 1,
} PxAnimationDirection;
```

Also, all enum members should have an assigned value. I forgot to add that on the phoenix side but I'll do that later. For now, just number them sequentially from 0, please.

lmichaelis reviewed (changes requested):
Thanks! I'm unsure about the `pxMdsGetAnimation_` scheme but it's fine for now. There are some formatting issues but I'll fix them myself later :) Also, you need to cast the return value of `.size()`-calls to `uint32_t` explicitly. Otherwise the compilers will complain.

JaXt0r commented:
Thanks for the feedback, @lmichaelis . I'll take care.

One question about the linter: Is there a way clang-lint can tell us what it complains about and not just where? I want to comply with it's demands, but never know what it is about as I also don't use it locally. 😉

lmichaelis commented:
> Thanks for the feedback, @lmichaelis . I'll take care.
> 
> One question about the linter: Is there a way clang-lint can tell us what it complains about and not just where? I want to comply with it's demands, but never know what it is about as I also don't use it locally. wink

I can look into it. Most likely it'll just be a diff showing what needs to be changed. Hmmm maybe I'll make it automatically add a commit in pull requests ..

JaXt0r committed:
Optimized: casting of size, renaming enum values, removing .editorconfig.

JaXt0r commented:
I took care of your requested changed. And tested it.

The only thing is again: clang-lint. :-) 
If you can at least share with me the lint settings, then I can take care of it myself.

One idea: Do you require CRLF? (As my 2 files are LF).

lmichaelis commented on include/phoenix/cffi/ModelScript.h:
```diff
@@ -14,7 +14,168 @@ typedef phoenix::model_script PxModelScript;
+typedef enum : uint8_t {
+	PxAnimationDirectionForward = 0,  ///< The animation samples are played from first to last.
+	PxAnimationDirectionBackward = 1, ///< The animation samples are played from last to first.
+} PxAnimationDirection;
```
C does not support typed enums. Remove `: uint8_t`.

lmichaelis reviewed (changes requested):
> If you can at least share with me the lint settings, then I can take care of it myself.

The `.clang-format`-file contains the style definitions.

> One idea: Do you require CRLF? (As my 2 files are LF).

Everything in Git should always be LF only (and that's the case here).

JaXt0r committed:
typedef enum fixed for C.

JaXt0r committed:
clang-formatting improved.

JaXt0r committed:
final clang fixes.

JaXt0r committed:
final clang fixes.

JaXt0r committed:
Check if clang-format prints the replacements it demands.

JaXt0r committed:
clang format executed locally...

JaXt0r committed:
final clang fixes.

JaXt0r committed:
bugfix

JaXt0r commented:
Hi  @lmichaelis 

I fixed all your code requests.

I also executed clang-lint locally and fixed a lot of formatting.
Now it fails on one additional method where I can't see why.

One idea: The method itself has an opening bracket not directly at 4-spaces/tab. Maybe this is the reason?

This is by the way the best bet I found on how to get more details about clang-format. But it didn't help me unfortunately:
https://discourse.llvm.org/t/make-clang-format-give-detailed-error-when-there-is-a-format-problem/5124/4

Please help!

JaXt0r committed:
clang fixed according to lmichaelis recommendation. ;-)

lmichaelis committed:
Formatting changes
